File: sdks/python/src/cvfile/detect.py
"""is_cv_file(): cheap probe for whether bytes look like a .cv file."""

from __future__ import annotations

import io

from pypdf import PdfReader
from pypdf.errors import PdfReadError

from cvfile._constants import CV_NAMESPACE_URI
from cvfile._pdf import read_metadata_xml


def is_cv_file(data: bytes) -> bool:
    if len(data) < 4 or data[:4] != b"%PDF":
        return False
    try:
        reader = PdfReader(io.BytesIO(data))
        xml = read_metadata_xml(reader)
        return xml is not None and CV_NAMESPACE_URI in xml
    except (PdfReadError, ValueError, KeyError):
        return False


__all__ = ["is_cv_file"]

File: sdks/go/cv/detect.go
package cv

import "strings"

// IsCvFile returns true if the bytes look like a valid .cv file.
func IsCvFile(data []byte) bool {
	if len(data) < 4 || string(data[:4]) != "%PDF" {
		return false
	}
	ctx, err := loadContext(data)
	if err != nil {
		return false
	}
	xml, err := readMetadataXML(ctx)
	if err != nil || xml == "" {
		return false
	}
	return strings.Contains(xml, NamespaceURI)
}
